Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -450,19 +458,18 @@ static void populateStencilOperation(MTLStencilDescriptor *mtlStencil, const WGP
     auto mtlFrontFace = frontFace(descriptor.primitive.frontFace);
     auto mtlCullMode = cullMode(descriptor.primitive.cullMode);
 
-    // FIXME: GPUPrimitiveState.unclippedDepth
-
     MTLRenderPipelineReflection *reflection;
-    const auto& pipelineLayout = WebGPU::fromAPI(descriptor.layout);
-    bool hasBindGroups = pipelineLayout.numberOfBindGroupLayouts() > 0;
-    id<MTLRenderPipelineState> renderPipelineState = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor options: hasBindGroups ? MTLPipelineOptionNone : MTLPipelineOptionArgumentInfo reflection:&reflection error:nil];
+    const PipelineLayout* pipelineLayout = nullptr;
+    if (descriptor.layout) {
+        if (auto& layout = WebGPU::fromAPI(descriptor.layout); layout.numberOfBindGroupLayouts())
+            pipelineLayout = &layout;
+    }
```

Comment:
This seems unrelated

Reply:
It is to avoid two `::create` functions with almost identical parameters